plugins/ata/fu-ata-device.c

```diff
@@ -474,19 +474,20 @@ fu_ata_device_activate (FuDevice *device, GError **error)
 	struct ata_tf tf = { 0x0 };
 
 	/* flush caches and make sure drive is ready to activate */
-	tf.dev = 0xa0 | ATA_USING_LBA;
+	tf.dev = ATA_USING_LBA;
 	tf.command = ATA_OP_STANDBY_IMMEDIATE;
-	if (!fu_ata_device_command (self, &tf, SG_DXFER_TO_DEV,
+	if (!fu_ata_device_command (self, &tf, SG_DXFER_NONE,
 				    120 * 1000, /* a long time! */
 				    NULL, 0, error)) {
 		g_prefix_error (error, "failed to standby immediate: ");
 		return FALSE;
 	}
 
 	/* load the new firmware */
+	tf.dev = 0x0a | ATA_USING_LBA;
 	tf.command = ATA_OP_DOWNLOAD_MICROCODE;
 	tf.feat = ATA_SUBCMD_MICROCODE_ACTIVATE;
-	if (!fu_ata_device_command (self, &tf, SG_DXFER_TO_DEV,
+	if (!fu_ata_device_command (self, &tf, SG_DXFER_NONE,
```

Comment:
I wonder why my drives didn't error for *_TO_DEV*? maybe they just didn't care?

Reply:
yeah maybe their firmware didn't mind.  I guess make sure this still works on yours though too :)

Reply:
Yup, will do tomorrow, wrong seat at the moment.